src/program/latex/generation/diagram/tikz.rs

```diff
@@ -0,0 +1,80 @@
+use std::fmt::format;
+
+pub mod quantikz;
+
+pub trait Tikz {
+    fn generate_default_quantikz_diagram(&self);
+}
+
+pub enum TikzOperator {
+    TikzLeftKet(u32),
+    TikzControl(i32),
+    TikzCnotTarget,
+    TikzCphaseTarget,
+    TikzSwap(i32),
+    TikzSwapTarget,
+    TikzNop,
+    TikzMeasure,
+}
+
+impl TikzOperator {
```

Comment:
This might be better off as an `impl fmt::Display for TikzOperator` block, which gives you an implementation of `ToString` for free